onnxruntime/test/providers/cpu/model_tests.cc

```diff
@@ -990,7 +993,43 @@ ::std::vector<::std::basic_string<ORTCHAR_T>> GetParameterStrings() {
   return v;
 }
 
-INSTANTIATE_TEST_SUITE_P(ModelTests, ModelTest, testing::ValuesIn(GetParameterStrings()));
+auto ExpandModelName  = [](const ::testing::TestParamInfo<ModelTest::ParamType>& info) {
+  // use info.param here to generate the test suffix
+  std::basic_string<ORTCHAR_T> name = info.param;
+
+  // the original name here is the combination of provider name and model path name
+  // remove the trailing 'xxxxxxx/model.onnx' of name
+  if (name.size() > 11 && name.substr(name.size() - 11) == ORT_TSTR("/model.onnx")) {
+    name = name.substr(0, info.param.size() - 11);
+  }
+  // remove the trailing 'xxxxxx.onnx' of name
+  else if (name.size() > 5 && name.substr(name.size() - 5) == ORT_TSTR(".onnx")) {
+    name = name.substr(0, info.param.size() - 5);
+  }
+
+  // Note: test name only accepts '_' and alphanumeric
+  // replace '/' or '\' with '_'
+  std::replace(name.begin(), name.end(), '/', '_');
+  std::replace(name.begin(), name.end(), '\\', '_');
+
+  // Note: test name only accepts '_' and alphanumeric
+  // remove '.' and '-'
+  char chars[] = ".-";
+  for (unsigned int i = 0; i < strlen(chars); ++i) {
+    name.erase(std::remove(name.begin(), name.end(), chars[i]), name.end());
+  }
+#ifdef _WIN32
+  // Note: The return value of INSTANTIATE_TEST_SUITE_P accpets std::basic_string<char...>.
+  // Need conversion of wchar_t to char.
+  return std::wstring_convert<std::codecvt_utf8<wchar_t>>().to_bytes(name);
```

Comment:
Just a tiny comment: internally ORT should only use UTF-8, but when you pass a string outside, for example, feed it to std::cout or a third-party library, you should use a local code page(like ISO-8859-* , GBK), not UTF-8.
Right now our model names do not contain non-ASCII string. In the future it won't too. So, it's fine to  use UTF-8 here. We don't need to make it support non-ASCII strings.